arbnode/batch_poster.go

```diff
@@ -440,6 +447,31 @@ func (b *BatchPoster) encodeAddBatch(seqNum *big.Int, prevMsgNum arbutil.Message
 }
 
 func (b *BatchPoster) estimateGas(ctx context.Context, sequencerMessage []byte, delayedMessages uint64) (uint64, error) {
+	config := b.config()
+	callOpts := &bind.CallOpts{
+		Context: ctx,
+	}
+	if config.DataPoster.WaitForL1Finality {
+		callOpts.BlockNumber = big.NewInt(int64(rpc.SafeBlockNumber))
+	}
+	safeDelayedMessagesBig, err := b.bridge.DelayedMessageCount(callOpts)
+	if err != nil {
+		return 0, err
+	}
+	if !safeDelayedMessagesBig.IsUint64() {
+		return 0, fmt.Errorf("calling delayedMessageCount() on the bridge returned a non-uint64 result %v", safeDelayedMessagesBig)
+	}
+	safeDelayedMessages := safeDelayedMessagesBig.Uint64()
```

Comment:
It seems like you can avoid getting delayedMessages as param for this function altogether and always use safeDelayedMessages for gas estimation.

Reply:
We could, but it's easy to pass the delayedMessages in, and I wanted to avoid a race condition if the sequencer was sequencing delayed messages faster than the batch poster expected (consider, for instance, if they had different L1s with different notions of "safe", since that's up to the consensus client to implement and isn't spec'd afaik).